src/Tasks.js

```diff
@@ -0,0 +1,139 @@
+    const sortOptions = ['title', 'created', 'priority']
+    const [sortBy, setSortBy] = useState(sortOptions[0]);
+    const [sortDir, setSortDir] = useState('asc');
+    const q = query(collection(db, collectionName), orderBy(sortBy, sortDir));   
+    const [taskList, loading, error] = useCollectionData(q);
+    const [showCompleted, setShowCompleted] = useState(true);
+
+    function onToggleComplete(){
+        setShowCompleted(!showCompleted)
+    }
+
+    function onChangeSortOption(index) {
+        setSortBy(sortOptions[index])
+    }
+
+    function onChangeSortDirection() {
+        if (sortDir==='asc') {
+            setSortDir('desc')
+        }else {
+            setSortDir('asc')
+        }
+    }
+
+    function deleteSingle(id){
+        deleteDoc(doc(db, collectionName, id))
+    }
+
+    function handleEditTask(id, field, value) {
+        updateDoc(doc(db, collectionName, id), {
+            [field]: value
+        })
+    }
+
+    function handleNewTaskSubmit(title, priority){
+        const taskId = generateUniqueID()
+        setDoc(doc(db, collectionName, taskId), {
+            id: taskId,
+            title: title,
+            completed: false,
+            created: serverTimestamp(),
+            priority: priority
+        })
+    }
+
+    function handleDeleteFinished(){
+        let tasksToDisplay = [...taskList]
+        let completed = tasksToDisplay.filter(task => task.completed)
+        completed.forEach(task => deleteDoc(doc(db, collectionName, task.id)))
+    }
+
+    const TasksToDisplay = () => {
+        const allTasks = [...taskList]
+        const completed = allTasks.filter(task => task.completed)
+        const uncompleted = allTasks.filter(task => !task.completed)
+
+        const tasksToDisplay = showCompleted ? uncompleted.concat(completed) : uncompleted
+
+        console.log("allTasks", allTasks)
+        console.log("sortBy", sortBy)
+
+        return (
+            <>
+            {tasksToDisplay.length===0 && (
+                <Typography>No tasks for display :)</Typography>
+            )}
+            {tasksToDisplay.length!==0 && (
+                tasksToDisplay.map(task => (
+                <Task
+                    key ={task.id}
+                    {...task}
+                    deleteSingle={deleteSingle}
+                    handleEditTask={handleEditTask}
+                />
+            )))}
+            </>
+        )
+    }
+
+    if (loading) {
+        return (
+            <>
+                <Navbar
+                    handleNewTaskSubmit={handleNewTaskSubmit}
+                    showCompleted={showCompleted}
+                    onToggleComplete={onToggleComplete}
+                    handleDeleteFinished={handleDeleteFinished}
+                    onChangeSortOption={onChangeSortOption}
+                    onChangeSortDirection={onChangeSortDirection}
+                />
+                <p>Loading</p>
+            </>
+        )
+    }else if (!error){
```

Comment:
Should display something in case of an error. Otherwise, user will be left with a blank screen.